arroyo-state/src/tables/key_time_multi_map.rs

```diff
@@ -124,10 +151,24 @@ impl<K: Key, V: Data> KeyTimeMultiMapCache<K, V> {
                         .push(tuple.value.unwrap());
                 }
                 DataOperation::DeleteKey => {
-                    values
-                        .entry(tuple.key)
-                        .or_default()
-                        .remove(&tuple.timestamp);
+                    values.remove(&tuple.key);
+                }
+                DataOperation::DeleteValue => {
+                    values.entry(tuple.key).and_modify(|map| {
+                        map.entry(tuple.timestamp).and_modify(|values| {
+                            values.retain(|value| value != &tuple.value.clone().unwrap());
```

Comment:
In general I think value-deletes should only remove a single value, even if there are more matches. For instance, if my state is keyed by campaign_id and has user_id, you could imagine a table of visits like

| campaign_id | user_id|
| 1 | 2 |
| 1 | 2 |
| 1 | 3 |
| 1 | 4 |

In this case (delete((1,2))) should only retract a single row.